restli-tools/src/main/java/com/linkedin/restli/tools/compatibility/ResourceCompatibilityChecker.java

```diff
@@ -224,6 +225,20 @@ private boolean checkEqualSingleValue(RecordDataSchema.Field field, Object prevD
     return true;
   }
 
+  private boolean checkD2ServiceName(String prevData, String currData) {
```

Comment:
what if `prevData` is null, but `currData` is set to the same as `resource name`?

Reply:
So I don't want to enforce that it should be the same as the resource name. That is a convention that people may not always choose to follow. For the purposes of GraphQL routing alone, I will add a check that if null, use the resource name, since we are confident about that convention there.

Reply:
Resolved offline with @mchen07